src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.Patch.cs

```diff
@@ -185,6 +185,10 @@ public abstract partial class InMemoryDocumentSessionOperations
             var pathScript = path.CompileToJavascript(_javascriptCompilationOptions.Value);
```

Comment:
maybe we should rename `_javascriptCompilationOptions` to `_pathScriptCompilationOptions` or something like that?
so that it will be more clear that these compilation options are only used for converting the `path` expression to js (and not the value)